src/main/java/com/volmit/iris/core/command/pregen/CommandIrisPregenCreate.java

```diff
@@ -0,0 +1,264 @@
+import com.volmit.iris.util.math.Position2;
+import com.volmit.iris.util.plugin.MortarCommand;
+import com.volmit.iris.util.plugin.VolmitSender;
+import org.bukkit.Bukkit;
+import org.bukkit.World;
+
+import java.util.Arrays;
+
+
+public class CommandIrisPregenCreate extends MortarCommand {
+
+    private static final KList<String> argus = new KList<>("radius=", "width=", "height=", "x=", "z=");
+
+    public CommandIrisPregenCreate() {
+        super("create", "c", "new", "+");
+        requiresPermission(Iris.perm);
+        setCategory("Pregen");
+        setDescription("""
+                Create a new pregeneration task.
+                Command usage & examples:
+                /iris pregen create [radius=<radius>] [width=<width>] [height=<height>] [x=<centerX>] [z=<centerZ>] [world=<world>] [-here]
+                /iris pregen create radius=5000 x=10r z=10r world=IrisWorld
+                /iris pregen create 5k -here
+                
+                <radius>:  Sets both width and height to a value.
+                <x> & <z>: Give the center point of the pregeneration.
+                <world>:   Specify a specific world name for generation as well (required for console)
+                -here:     If added, the center location is set to your position (player only)
+                
+                For all numeric values (radius, centerX, etc.) you may use:
+                c => 16, r => 512, k => 1000
+                Example: entering '1000' is the same as '1k' (1 * 1000)
+                https://docs.volmit.com/iris/pregeneration""");
+    }
+
+    @Override
+    public void addTabOptions(VolmitSender sender, String[] args, KList<String> list) {
+
+        // Add arguments
+        argus.forEach(p -> {
+            boolean hasArg = false;
+            for (String arg : args) {
+                if (!arg.contains("=") || !p.contains("=")){
+                    continue;
+                }
+                if (arg.split("=")[0].equals(p.split("=")[0])) {
+                    hasArg = true;
+                    break;
+                }
+            }
+            if (!hasArg){
+                list.add(p);
+            }
+        });
+
+        // Add -here
+        boolean hasHere = false;
+        for (String arg : args) {
+            if (arg.equals("-here")) {
+                hasHere = true;
+                break;
+            }
+        }
+        if (!hasHere){
+            list.add("-here");
+        }
+
+        // Add Iris worlds
+        if (Bukkit.getWorlds().isEmpty()){
+            list.add("world=<name>");
+        } else {
+            Bukkit.getWorlds().forEach(w -> {
+                if (IrisToolbelt.isIrisWorld(w)) {
+                    list.add("world=" + w.getName());
+                }
+            });
+        }
+    }
+
+    @Override
+    protected String getArgsUsage() {
+        return "<radius> [width=<width>] [height=<height>] [x=<centerX>] [z=<centerZ>] [world=<world>] [-here]";
```

Comment:
The width & height setting bothers me a lot. Im not confident it's fully supported. It would be safe to rely on a size only. That and no one will really use this as generally pregens are symetrical. And ... with this they now have to type out two long arguments instead of just a size.

Reply:
You can still do /iris pregen create 5000
Or /iris pregen create radius=5000
Same thing.
Tried the width&height with different values and it seems to work
I can remove them from the description and args usage but keep functionality (hiding them but still allowing usage), or I can just keep them, up 2 u.